src/main/java/fi/aalto/cs/apluscourses/utils/CoursesClient.java

```diff
@@ -289,4 +292,30 @@ private static void requireResponseEntity(@NotNull HttpResponse response)
       throw new UnexpectedResponseException(response, "Response is missing body");
     }
   }
+
+  @NotNull
+  private static CloseableHttpClient createClient() {
+    return HttpClients.custom().setUserAgent(getUserAgent()).build();
```

Comment:
I wonder if `custom` also changes other things from `createDefault`, such as timeouts or something else

Reply:
An alternative would be to add the user agent to the request instead of the client

Reply:
Correct. I was wondering the same thing when coding this; I couldn't find no clear documentation but look at the decompiled code reveals they are the same thing:

```java
public class HttpClients {
  // ...
  public static HttpClientBuilder custom() {
    return HttpClientBuilder.create();
  }

  public static CloseableHttpClient createDefault() {
    return HttpClientBuilder.create().build();
  }
  // ...
}

Reply:
Nice find! It's good as is in that case